cinema/tests/test_movie_api.py

```diff
@@ -0,0 +1,143 @@
+from django.test import TestCase
+
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from cinema.models import Movie, Genre, Actor
+
+
+class MovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.drama = Genre.objects.create(
+            name="Drama",
+        )
+        self.comedy = Genre.objects.create(
+            name="Comedy",
+        )
+        self.actress = Actor.objects.create(first_name="Kate", last_name="Winslet")
+        self.movie = Movie.objects.create(
+            title="Titanic",
+            description="Titanic description",
+            duration=123,
+        )
+        self.movie.genres.add(self.drama)
+        self.movie.genres.add(self.comedy)
+        self.movie.actors.add(self.actress)
+
+    def test_get_movies(self):
+        movies = self.client.get("/api/cinema/movies/")
+        titanic = {
+            "title": "Titanic",
+            "description": "Titanic description",
+            "duration": 123,
+            "genres": ["Drama", "Comedy"],
+            "actors": ["Kate Winslet"],
+        }
+        print(movies.data)
+        self.assertEqual(movies.status_code, status.HTTP_200_OK)
+        for field in titanic:
+            self.assertEqual(movies.data[0][field], titanic[field])
+
+    def test_get_movies_with_genre_filtering(self):
+        movies = self.client.get(f"/api/cinema/movies/?genres={self.comedy.id}")
+        self.assertEqual(len(movies.data), 1)
+        movies = self.client.get(f"/api/cinema/movies/?genres={123213}")
+        self.assertEqual(len(movies.data), 0)
```

Comment:
please add check for filtering with multiple genres & multiple actors
`?genres=1,2`

Reply:
Done